### DISCUSSION TITLE
Unable to build docker image.

### DISCUSSION THREAD

**Discussion by shrey-creator** [pacexy/flow#84] [Q&A]
Hey ,
I am trying to build docker image by running and it is failing at the following step -
RUN pnpm i --frozen-lockfile                                                                                         

Following is the error message - > 
=> ERROR [reader installer  9/13] RUN pnpm i --frozen-lockfile                                                                               91.7s 
------
 > [reader installer  9/13] RUN pnpm i --frozen-lockfile:
20.83 Scope: all 5 workspace projects
21.02 Lockfile is up to date, resolution step is skipped
21.12 Progress: resolved 1, reused 0, downloaded 0, added 0
21.59 Packages: +1745
21.59 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
21.62  WARN  GET https://registry.npmjs.org/@types/localforage/-/localforage-0.0.34.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.63  WARN  GET https://registry.npmjs.org/localforage/-/localforage-1.10.0.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.21.63  WARN  GET https://registry.npmjs.org/lie/-/lie-3.1.1.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.63  WARN  GET https://registry.npmjs.org/immediate/-/immediate-3.0.6.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.     
21.63  WARN  GET https://registry.npmjs.org/@xmldom/xmldom/-/xmldom-0.7.5.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.   
21.63  WARN  GET https://registry.npmjs.org/core-js/-/core-js-3.21.1.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.        
21.64  WARN  GET https://registry.npmjs.org/event-emitter/-/event-emitter-0.3.5.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.64  WARN  GET https://registry.npmjs.org/d/-/d-1.0.1.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.64  WARN  GET https://registry.npmjs.org/es5-ext/-/es5-ext-0.10.60.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.       
21.64  WARN  GET https://registry.npmjs.org/es6-iterator/-/es6-iterator-2.0.3.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.64  WARN  GET https://registry.npmjs.org/es6-symbol/-/es6-symbol-3.1.3.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.   
21.65  WARN  GET https://registry.npmjs.org/ext/-/ext-1.6.0.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.65  WARN  GET https://registry.npmjs.org/type/-/type-2.6.0.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.65  WARN  GET https://registry.npmjs.org/next-tick/-/next-tick-1.1.0.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.     
21.65  WARN  GET https://registry.npmjs.org/type/-/type-1.2.0.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
21.65  WARN  GET https://registry.npmjs.org/jszip/-/jszip-3.10.1.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
22.14 Progress: resolved 1745, reused 0, downloaded 0, added 0
31.63  WARN  GET https://registry.npmjs.org/@types/localforage/-/localforage-0.0.34.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries 
left.
31.63  WARN  GET https://registry.npmjs.org/localforage/-/localforage-1.10.0.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.  
31.63  WARN  GET https://registry.npmjs.org/lie/-/lie-3.1.1.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.64  WARN  GET https://registry.npmjs.org/immediate/-/immediate-3.0.6.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.       
31.64  WARN  GET https://registry.npmjs.org/@xmldom/xmldom/-/xmldom-0.7.5.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.     
31.64  WARN  GET https://registry.npmjs.org/core-js/-/core-js-3.21.1.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.64  WARN  GET https://registry.npmjs.org/event-emitter/-/event-emitter-0.3.5.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.64  WARN  GET https://registry.npmjs.org/d/-/d-1.0.1.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.64  WARN  GET https://registry.npmjs.org/es5-ext/-/es5-ext-0.10.60.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.64  WARN  GET https://registry.npmjs.org/es6-iterator/-/es6-iterator-2.0.3.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left. 
31.64  WARN  GET https://registry.npmjs.org/es6-symbol/-/es6-symbol-3.1.3.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.     
31.65  WARN  GET https://registry.npmjs.org/ext/-/ext-1.6.0.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.65  WARN  GET https://registry.npmjs.org/type/-/type-2.6.0.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.65  WARN  GET https://registry.npmjs.org/next-tick/-/next-tick-1.1.0.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.       
31.65  WARN  GET https://registry.npmjs.org/type/-/type-1.2.0.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
31.65  WARN  GET https://registry.npmjs.org/jszip/-/jszip-3.10.1.tgz error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
91.65  WARN  GET https://registry.npmjs.org/lie/-/lie-3.3.0.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.66  ERR_INVALID_THIS  Value of "this" must be of type URLSearchParams
91.66
91.66 pnpm [ERR_INVALID_THIS]: Value of "this" must be of type URLSearchParams
91.66     at Proxy.getAll (node:internal/url:533:13)
91.66     at Proxy.<anonymous> (/root/.cache/node/corepack/pnpm/7.10.0/dist/pnpm.cjs:52878:55)
91.66     at /root/.cache/node/corepack/pnpm/7.10.0/dist/pnpm.cjs:52931:31
91.66     at Array.reduce (<anonymous>)
91.66     at Proxy.raw (/root/.cache/node/corepack/pnpm/7.10.0/dist/pnpm.cjs:52930:33)
91.66     at new Headers (/root/.cache/node/corepack/pnpm/7.10.0/dist/pnpm.cjs:52824:28)
91.66     at getNodeRequestOptions (/root/.cache/node/corepack/pnpm/7.10.0/dist/pnpm.cjs:53143:23)
91.66     at /root/.cache/node/corepack/pnpm/7.10.0/dist/pnpm.cjs:53200:25
91.66     at new Promise (<anonymous>)
91.66     at fetch (/root/.cache/node/corepack/pnpm/7.10.0/dist/pnpm.cjs:53198:14)
91.66  WARN  GET https://registry.npmjs.org/pako/-/pako-1.0.11.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.66  WARN  GET https://registry.npmjs.org/readable-stream/-/readable-stream-2.3.7.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.67  WARN  GET https://registry.npmjs.org/core-util-is/-/core-util-is-1.0.3.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.67  WARN  GET https://registry.npmjs.org/inherits/-/inherits-2.0.4.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.       
91.67  WARN  GET https://registry.npmjs.org/isarray/-/isarray-1.0.0.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.67  WARN  GET https://registry.npmjs.org/process-nextick-args/-/process-nextick-args-2.0.1.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.67  WARN  GET https://registry.npmjs.org/safe-buffer/-/safe-buffer-5.1.2.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left. 
91.68  WARN  GET https://registry.npmjs.org/string_decoder/-/string_decoder-1.1.1.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries 
left.
91.68  WARN  GET https://registry.npmjs.org/util-deprecate/-/util-deprecate-1.0.2.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries 
left.
91.68  WARN  GET https://registry.npmjs.org/setimmediate/-/setimmediate-1.0.5.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.68  WARN  GET https://registry.npmjs.org/lodash/-/lodash-4.17.21.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.69  WARN  GET https://registry.npmjs.org/marks-pane/-/marks-pane-1.0.9.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.   
91.69  WARN  GET https://registry.npmjs.org/path-webpack/-/path-webpack-0.0.3.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.69  WARN  GET https://registry.npmjs.org/@babel/cli/-/cli-7.19.3.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
91.69  WARN  GET https://registry.npmjs.org/@ampproject/remapping/-/remapping-2.1.2.tgz error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
------
failed to solve: process "/bin/sh -c pnpm i --frozen-lockfile" did not complete successfully: exit code: 1

Can anyone help?

**Comment by Chay751** [comment]
Edit docker file
`FROM node:alpine AS installer`
Modified to:
`FROM node:16-alpine AS installer`

`FROM node:alpine AS runner`
Modified to:
`FROM node:16-alpine AS runner`

and then buiild img will work
（It should be a node version issue）

**Comment by SongRunqi** [comment]
I met the same problem

**Comment by Chay751** [comment]
您好，已收到邮件！
